web/src/settings_profile_fields.js

```diff
@@ -599,6 +603,31 @@ function toggle_display_in_profile_summary_profile_field(e) {
     );
 }
 
+function toggle_required(e) {
+    const field_id = Number.parseInt($(e.currentTarget).attr("data-profile-field-id"), 10);
+    const field = get_profile_field(field_id);
+
+    let field_data;
+    if (field.field_data) {
+        field_data = field.field_data;
+    }
+
+    const data = {
+        name: field.name,
+        hint: field.hint,
+        field_data,
+        display_in_profile_summary: field.display_in_profile_summary,
+        required: !field.required,
+    };
```

Comment:
Do you need to pass all these fields, or could it just pass the one field we're changing?

Reply:
It appears the answer is yes; kinda a weird quirk in how the custom profile fields API works compared to the rest of the Zulip API, but I don't think we need to try to change that as part of this PR.